h/search/core.py

```diff
@@ -44,6 +44,7 @@ def __init__(self, request, separate_replies=False, stats=None, _replies_limit=2
         self._modifiers = [query.Sorter(),
                            query.Limiter(),
                            query.DeletedFilter(),
+                           query.HiddenFilter(),
```

Comment:
Thinking out loud here; I'm a little brain dead today but I think this is indeed where we want this to go as we want this logic to happen on activity pages, sidebar, feeds, api search, badge, etc. If we only wanted this logic on the sidebar then, instead of adding it here, we would add it into the h/views/api/annotations and possibly the h/views/feeds.